rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -287,25 +278,27 @@ pub(crate) fn run(
 
                     // In a normal Rust application, Sentry's guard would flush on drop: https://docs.sentry.io/platforms/rust/configuration/draining/
                     // But due to a limit in `tao` we cannot return from the event loop and must call `std::process::exit` (or Tauri's wrapper), so we explicitly flush here.
+                    // TODO: This limit may not exist in Tauri v2
                     telemetry.stop();
 
                     tracing::info!(?exit_code);
                     app_handle.exit(exit_code);
+                    // In Tauri v1, calling `App::exit` internally exited the process.
+                    // In Tauri v2, that doesn't happen, but `App::run` still doesn't return, so we have to bail out of the process manually.
+                    std::process::exit(exit_code);
                 });
                 Ok(())
             };
 
-            setup_result_tx.send(setup_inner()).expect("should be able to send setup result");
+            if let Err(error) = setup_inner() {
+                firezone_telemetry::capture_anyhow(&error);
+                tracing::error!(?error, "Tauri setup failed");
```

Comment:
Please update according to https://github.com/firezone/firezone/pull/7104 and https://github.com/firezone/firezone/pull/7105.

Reply:
Why don't we return an error from this entire function instead?